### PR TITLE
[14.0] edi_sale_oca: use edi.configuration 

### PR DESCRIPTION
**Pull request by simahawk**
Depends on
- [x] #1035

* provides full support and examples on how to use edi.conf in real life
* after we merge this we can get rid of exc type auto in #1024

### CODE REVIEW

**Comment by simahawk** [comment]
@JordiMForgeFlow would you mind to have a look at this?
I'll work on sale ubl and purchase ubl as well in the same fashion (using new edi.configuration model) and you'll likely benefit from it in v17 ;)

File: edi_sale_oca/demo/edi_configuration.xml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        <field name="trigger_id" ref="edi_conf_trigger_sale_order_state_change" />
+        <field name="snippet_do">
+# STATES
+# ('draft', 'Quotation'),
+# ('sent', 'Quotation Sent'),
+# ('sale', 'Sales Order'),
+# ('done', 'Locked'),
+# ('cancel', 'Cancelled'),
+if record.state == 'sale':
+  record._edi_send_via_edi(conf.type_id)
```
Comment (SilvioC2C) [inline edi_sale_oca/demo/edi_configuration.xml:20]: Could be useful to add help text explaining which variables are available to use in the snippet (like Odoo does by default for server actions or crons)

**Review by SilvioC2C — APPROVED** [review]
Overall LGTM, one small imrpovement suggested

File: edi_sale_oca/demo/edi_configuration.xml (hunk 1 above)
Comment (simahawk) [reply edi_sale_oca/demo/edi_configuration.xml:20]: variables come from edi_oca. I'll check there.

**Comment by simahawk** [comment]
Build on 14.0 is broken https://github.com/OCA/edi/issues/1084

**Pull request merged**